[Fleet] Improve package policy bulk upgrade performance

nchaulet opened the pull request:
## Summary

Resolve https://github.com/elastic/kibana/issues/205332

Improve package policy upgrade performance.

That PR improve the package policy bulk upgrade performance, for this:
* I refactorized our upgrade method to have a real `bulkUpgrade` method that use other `bulkGet` and `bulkUpdate` operations to limit the number of call to ES
* I tried to prefect packageInfo, assetsMaps, instead of fetching it for each package policies
* I tried to cache expensive operations using our request cache
* I used the already asyncDeploy in bulkUpdate to bump agent policies in a task after the upgrade, making the response a lot faster.


### Implementations details

#### Split `upgrade` and `bulkUpgrade`

`bulkUpgrade` use bulk operations and try to group and reuse the models that we need to fetch packageInfo, assetsMaps, ...

#### Changes to `bulkUpdate`

It seems `bulkUpdate` was lacking a few things that was done in regular `update`, I tried to fix that

##### Caching

* I added a little more caching, we have a cache store per request that I used here to cache:
  * If space awareness is enabled
  * Agent template compilation 


## Benchmark

First benchmark seems really encouraging with upgrade 100 aws package policies going from 100s to 5s

With 100 outdated aws agent policies

#### Before 
<img width="1263" alt="Screenshot 2025-03-07 at 1 12 47 PM" src="https://github.com/user-attachments/assets/0273f638-0f18-4731-92f0-6acbb5d119d2" />



#### After


<img width="1217" alt="Screenshot 2025-03-07 at 1 01 34 PM" src="https://github.com/user-attachments/assets/0eb56735-e6f0-4524-a596-47d74cae5ee4" />

nchaulet committed:
[Fleet] Improve bulk upgrade performance

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-cloud-from-pr --> Click to trigger **kibana-deploy-cloud-from-pr** for this PR!

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/281854)
* Commit: 7bbd4c516dc14952cacad8d36008c271d68ce548

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/281854#01957147-8225-41bf-862f-9d1ec5ebcd82)

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/281854","number":281854,"commit":"7bbd4c516dc14952cacad8d36008c271d68ce548"}],"number":281854}
buildkite-pr-comment-->

nchaulet commented on x-pack/platform/plugins/shared/fleet/server/services/epm/agent/agent.ts:
```diff
@@ -79,7 +90,7 @@ function buildTemplateVariables(logger: Logger, variables: PackagePolicyConfigRe
     const keyParts = key.split('.');
     const lastKeyPart = keyParts.pop();
-    logger.debug(`Building agent template variables`);
+    // logger.debug(`Building agent template variables`);
```
Todo un-uncomment

nchaulet commented on x-pack/platform/plugins/shared/fleet/server/services/epm/agent/agent.ts:
```diff
@@ -15,14 +15,25 @@ import { toCompiledSecretRef } from '../../secrets';
+    let template = getHandlebarsCompiledTemplateCache(templateStr);
+
+    if (!template) {
+      template = handlebars.compile(templateStr, { noEscape: true });
```
this seems a big change, as for 100 agent policies, this seems to be called at least 200 times per template.

nchaulet committed:
fix tests

nchaulet committed:
Merge branch 'main' of github.com:elastic/kibana into feature-bulk-upgrade

nchaulet committed:
fix tests

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/281939)
* Commit: 18b3186180e31a063aab97771d2604a5147ed647

### Failed CI Steps
* [Jest Tests #18](https://buildkite.com/elastic/kibana-pull-request/builds/281939#01957292-af1f-4017-86d6-8f68c0a6caab)
* [FTR Configs #10](https://buildkite.com/elastic/kibana-pull-request/builds/281939#01957292-af31-43ab-ad17-f67b94dc9560)
* [FTR Configs #10](https://buildkite.com/elastic/kibana-pull-request/builds/281939#019572b2-613e-4b42-bf8a-7a756340dad3)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/281939#01957292-af31-43ab-ad17-f67b94dc9560) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/281939/jobs/01957292-af31-43ab-ad17-f67b94dc9560/artifacts/019572b2-457b-46fc-8209-ac7c9e53b4b5) FTR Configs #<span></span>10 / Fleet Endpoints fleet policy secrets fleet server version requirements should store new secrets after package upgrade
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/281939#019572b2-613e-4b42-bf8a-7a756340dad3) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/281939/jobs/019572b2-613e-4b42-bf8a-7a756340dad3/artifacts/019572bd-3d1b-4e58-a5ed-2acb49d9e05a) FTR Configs #<span></span>10 / Fleet Endpoints fleet policy secrets fleet server version requirements should store new secrets after package upgrade
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/281939#01957292-af1f-4017-86d6-8f68c0a6caab) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/281939/jobs/01957292-af1f-4017-86d6-8f68c0a6caab/artifacts/019572b6-7aaa-424a-ab59-c9706596c998) Jest Tests #<span></span>18 / Package policy service delete should call audit logger
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/281939#01957292-af1f-4017-86d6-8f68c0a6caab) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/281939/jobs/01957292-af1f-4017-86d6-8f68c0a6caab/artifacts/019572b6-7aaa-4349-be3c-e0efcf01e67f) Jest Tests #<span></span>18 / When calling package policy upgrade api handler should return valid response

### History
* :broken_heart: [Build #281888](https://buildkite.com/elastic/kibana-pull-request/builds/281888) failed 30a62780c65f1a5016cc1b260aaccd57b337ca4a
* :broken_heart: [Build #281854](https://buildkite.com/elastic/kibana-pull-request/builds/281854) failed 7bbd4c516dc14952cacad8d36008c271d68ce548

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/281939","number":281939,"commit":"18b3186180e31a063aab97771d2604a5147ed647"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/281888","number":281888,"commit":"30a62780c65f1a5016cc1b260aaccd57b337ca4a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/281854","number":281854,"commit":"7bbd4c516dc14952cacad8d36008c271d68ce548"}],"number":281939}
buildkite-pr-comment-->

nchaulet committed:
fix tests

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/capture_oas_snapshot --include-path /api/status --include-path /api/alerting/rule/ --include-path /api/alerting/rules --include-path /api/actions --include-path /api/security/role --include-path /api/spaces --include-path /api/fleet --include-path /api/dashboards --update'

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/282295)
* Commit: 7fbc38dc54b669e08e48123f1a21342578dd9893

### Failed CI Steps
* [Check OAS Snapshot](https://buildkite.com/elastic/kibana-pull-request/builds/282295#0195809e-8843-42a8-836c-f9e9baf9cdec)

### History
* :broken_heart: [Build #281939](https://buildkite.com/elastic/kibana-pull-request/builds/281939) failed 18b3186180e31a063aab97771d2604a5147ed647
* :broken_heart: [Build #281888](https://buildkite.com/elastic/kibana-pull-request/builds/281888) failed 30a62780c65f1a5016cc1b260aaccd57b337ca4a
* :broken_heart: [Build #281854](https://buildkite.com/elastic/kibana-pull-request/builds/281854) failed 7bbd4c516dc14952cacad8d36008c271d68ce548

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/282295","number":282295,"commit":"7fbc38dc54b669e08e48123f1a21342578dd9893"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/281939","number":281939,"commit":"18b3186180e31a063aab97771d2604a5147ed647"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/281888","number":281888,"commit":"30a62780c65f1a5016cc1b260aaccd57b337ca4a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/281854","number":281854,"commit":"7bbd4c516dc14952cacad8d36008c271d68ce548"}],"number":282295}
buildkite-pr-comment-->

kibanamachine committed:
[CI] Auto-commit changed files from 'make api-docs'

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/282431)
* Commit: d46d96a82e3a439d04b3159d207c6c6982b296d7

### Failed CI Steps
* [FTR Configs #117](https://buildkite.com/elastic/kibana-pull-request/builds/282431#0195816d-c409-4daa-be89-2a4492231cb4)
* [FTR Configs #117](https://buildkite.com/elastic/kibana-pull-request/builds/282431#0195818b-4c8d-48c6-88c3-b937d6ae07d3)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/282431#0195816d-c409-4daa-be89-2a4492231cb4) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/282431/jobs/0195816d-c409-4daa-be89-2a4492231cb4/artifacts/0195818b-3dc8-4053-b8f9-34c1aaea27a4) FTR Configs #<span></span>117 / Fleet Endpoints fleet policy secrets fleet server version requirements should store new secrets after package upgrade
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/282431#0195818b-4c8d-48c6-88c3-b937d6ae07d3) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/282431/jobs/0195818b-4c8d-48c6-88c3-b937d6ae07d3/artifacts/01958195-4f32-45a0-b8ec-506277554e47) FTR Configs #<span></span>117 / Fleet Endpoints fleet policy secrets fleet server version requirements should store new secrets after package upgrade

### History
* :broken_heart: [Build #282319](https://buildkite.com/elastic/kibana-pull-request/builds/282319) failed 1712bce1b208bfff5f9b550deee5998f755775a3
* :broken_heart: [Build #282295](https://buildkite.com/elastic/kibana-pull-request/builds/282295) failed 7fbc38dc54b669e08e48123f1a21342578dd9893
* :broken_heart: [Build #281939](https://buildkite.com/elastic/kibana-pull-request/builds/281939) failed 18b3186180e31a063aab97771d2604a5147ed647
* :broken_heart: [Build #281888](https://buildkite.com/elastic/kibana-pull-request/builds/281888) failed 30a62780c65f1a5016cc1b260aaccd57b337ca4a
* :broken_heart: [Build #281854](https://buildkite.com/elastic/kibana-pull-request/builds/281854) failed 7bbd4c516dc14952cacad8d36008c271d68ce548

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/282431","number":282431,"commit":"d46d96a82e3a439d04b3159d207c6c6982b296d7"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/282319","number":282319,"commit":"1712bce1b208bfff5f9b550deee5998f755775a3"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/282295","number":282295,"commit":"7fbc38dc54b669e08e48123f1a21342578dd9893"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/281939","number":281939,"commit":"18b3186180e31a063aab97771d2604a5147ed647"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/281888","number":281888,"commit":"30a62780c65f1a5016cc1b260aaccd57b337ca4a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/281854","number":281854,"commit":"7bbd4c516dc14952cacad8d36008c271d68ce548"}],"number":282431}
buildkite-pr-comment-->

nchaulet committed:
fix tests

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/282495)
* Commit: d46d96a82e3a439d04b3159d207c6c6982b296d7

### Failed CI Steps
* [Jest Tests #15](https://buildkite.com/elastic/kibana-pull-request/builds/282495#01958241-3ac2-4d84-a406-a3abd9229d2a)
* [FTR Configs #12](https://buildkite.com/elastic/kibana-pull-request/builds/282495#01958241-3ad2-4dc0-ba0a-d2bd2021d68e)
* [FTR Configs #12](https://buildkite.com/elastic/kibana-pull-request/builds/282495#01958260-6ca9-493c-9cbf-f0c2752d5406)
* [FTR Configs #113](https://buildkite.com/elastic/kibana-pull-request/builds/282495#01958241-3b0f-4a92-8263-ee265c4d8111)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/282495#01958241-3b0f-4a92-8263-ee265c4d8111) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/282495/jobs/01958241-3b0f-4a92-8263-ee265c4d8111/artifacts/01958265-d3c3-4122-bf60-09e514e316d2) FTR Configs #<span></span>113 / @<span></span>ess @<span></span>serverless @<span></span>serverlessQA SecuritySolution Timeline Timeline @<span></span>skipInServerless resolve timeline "before all" hook for "should return outcome exactMatch when the id is unchanged"
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/282495#01958241-3ad2-4dc0-ba0a-d2bd2021d68e) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/282495/jobs/01958241-3ad2-4dc0-ba0a-d2bd2021d68e/artifacts/01958260-4e17-4f75-a87a-b61f2694f0a9) FTR Configs #<span></span>12 / Fleet Endpoints fleet policy secrets fleet server version requirements should store new secrets after package upgrade
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/282495#01958260-6ca9-493c-9cbf-f0c2752d5406) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/282495/jobs/01958260-6ca9-493c-9cbf-f0c2752d5406/artifacts/0195826a-dbf6-4f85-aa9f-b98f83fb4b61) FTR Configs #<span></span>12 / Fleet Endpoints fleet policy secrets fleet server version requirements should store new secrets after package upgrade
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/282495#01958241-3ac2-4d84-a406-a3abd9229d2a) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/282495/jobs/01958241-3ac2-4d84-a406-a3abd9229d2a/artifacts/01958263-57c4-4260-abd8-cd39b8af5f11) Jest Tests #<span></span>15 / Package policy service delete should call audit logger
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/282495#01958241-3ac2-4d84-a406-a3abd9229d2a) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/282495/jobs/01958241-3ac2-4d84-a406-a3abd9229d2a/artifacts/01958263-57c2-47fe-888e-ffec260f4be1) Jest Tests #<span></span>15 / When calling package policy dry run upgrade api handler should return valid response
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/282495#01958241-3ac2-4d84-a406-a3abd9229d2a) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/282495/jobs/01958241-3ac2-4d84-a406-a3abd9229d2a/artifacts/01958263-57c3-49dc-91f7-5e9149f29134) Jest Tests #<span></span>15 / When calling package policy upgrade api handler should return valid response

### History
* :broken_heart: [Build #282431](https://buildkite.com/elastic/kibana-pull-request/builds/282431) failed d46d96a82e3a439d04b3159d207c6c6982b296d7
* :broken_heart: [Build #282319](https://buildkite.com/elastic/kibana-pull-request/builds/282319) failed 1712bce1b208bfff5f9b550deee5998f755775a3
* :broken_heart: [Build #282295](https://buildkite.com/elastic/kibana-pull-request/builds/282295) failed 7fbc38dc54b669e08e48123f1a21342578dd9893
* :broken_heart: [Build #281939](https://buildkite.com/elastic/kibana-pull-request/builds/281939) failed 18b3186180e31a063aab97771d2604a5147ed647
* :broken_heart: [Build #281888](https://buildkite.com/elastic/kibana-pull-request/builds/281888) failed 30a62780c65f1a5016cc1b260aaccd57b337ca4a

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/282495","number":282495,"commit":"d46d96a82e3a439d04b3159d207c6c6982b296d7"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/282431","number":282431,"commit":"d46d96a82e3a439d04b3159d207c6c6982b296d7"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/282319","number":282319,"commit":"1712bce1b208bfff5f9b550deee5998f755775a3"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/282295","number":282295,"commit":"7fbc38dc54b669e08e48123f1a21342578dd9893"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/281939","number":281939,"commit":"18b3186180e31a063aab97771d2604a5147ed647"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/281888","number":281888,"commit":"30a62780c65f1a5016cc1b260aaccd57b337ca4a"}],"number":282495}
buildkite-pr-comment-->

nchaulet committed:
Merge branch 'feature-bulk-upgrade' of github.com:nchaulet/kibana into feature-bulk-upgrade

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/capture_oas_snapshot --include-path /api/status --include-path /api/alerting/rule/ --include-path /api/alerting/rules --include-path /api/actions --include-path /api/security/role --include-path /api/spaces --include-path /api/fleet --include-path /api/dashboards --update'

elasticmachine commented:
Pinging @elastic/fleet (Team:Fleet)

kibanamachine committed:
[CI] Auto-commit changed files from 'make api-docs'

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/282520)
* Commit: c285e316af12c2c1c0e3cac8773ee86a1e241316

### Failed CI Steps
* [Jest Tests #6](https://buildkite.com/elastic/kibana-pull-request/builds/282520#019582a6-9eba-4590-a97c-c03d07526d65)
* [Jest Tests #6](https://buildkite.com/elastic/kibana-pull-request/builds/282520#019582c9-dab7-4fad-a72b-3e1d3881c6c4)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/282520#019582a6-9eba-4590-a97c-c03d07526d65) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/282520/jobs/019582a6-9eba-4590-a97c-c03d07526d65/artifacts/019582c9-abe1-404a-92d1-ddb84d83595a) Jest Tests #<span></span>6 / Package policy service delete should call audit logger
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/282520#019582c9-dab7-4fad-a72b-3e1d3881c6c4) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/282520/jobs/019582c9-dab7-4fad-a72b-3e1d3881c6c4/artifacts/019582da-ddad-4e9b-91e4-44e4d624f22a) Jest Tests #<span></span>6 / Package policy service delete should call audit logger

### History
* :broken_heart: [Build #282495](https://buildkite.com/elastic/kibana-pull-request/builds/282495) failed d46d96a82e3a439d04b3159d207c6c6982b296d7
* :broken_heart: [Build #282431](https://buildkite.com/elastic/kibana-pull-request/builds/282431) failed d46d96a82e3a439d04b3159d207c6c6982b296d7
* :broken_heart: [Build #282319](https://buildkite.com/elastic/kibana-pull-request/builds/282319) failed 1712bce1b208bfff5f9b550deee5998f755775a3
* :broken_heart: [Build #282295](https://buildkite.com/elastic/kibana-pull-request/builds/282295) failed 7fbc38dc54b669e08e48123f1a21342578dd9893
* :broken_heart: [Build #281939](https://buildkite.com/elastic/kibana-pull-request/builds/281939) failed 18b3186180e31a063aab97771d2604a5147ed647
* :broken_heart: [Build #281888](https://buildkite.com/elastic/kibana-pull-request/builds/281888) failed 30a62780c65f1a5016cc1b260aaccd57b337ca4a

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/282520","number":282520,"commit":"c285e316af12c2c1c0e3cac8773ee86a1e241316"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/282495","number":282495,"commit":"d46d96a82e3a439d04b3159d207c6c6982b296d7"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/282431","number":282431,"commit":"d46d96a82e3a439d04b3159d207c6c6982b296d7"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/282319","number":282319,"commit":"1712bce1b208bfff5f9b550deee5998f755775a3"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/282295","number":282295,"commit":"7fbc38dc54b669e08e48123f1a21342578dd9893"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/281939","number":281939,"commit":"18b3186180e31a063aab97771d2604a5147ed647"}],"number":282520}
buildkite-pr-comment-->

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-cloud-from-pr --> Click to trigger **kibana-deploy-cloud-from-pr** for this PR!

nchaulet committed:
fix tests

nchaulet committed:
Merge branch 'feature-bulk-upgrade' of github.com:nchaulet/kibana into feature-bulk-upgrade

elasticmachine committed:
Merge branch 'main' into feature-bulk-upgrade

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/282770)
* Commit: ac6bd5af605d1fe03706cf4ef84c06b976b835a1

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Public APIs missing comments
> Total count of every public API that lacks a comment. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats comments` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/e0bf98e45a6c2b4e138cac601c6082ca68fb8301) | [after](https://github.com/elastic/kibana/commit/ac6bd5af605d1fe03706cf4ef84c06b976b835a1) | diff |
| --- | --- | --- | --- |
| `fleet` | 1338 | 1346 | +8 |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/e0bf98e45a6c2b4e138cac601c6082ca68fb8301) | [after](https://github.com/elastic/kibana/commit/ac6bd5af605d1fe03706cf4ef84c06b976b835a1) | diff |
| --- | --- | --- | --- |
| `fleet` | 1466 | 1474 | +8 |


</details>


### History
* :broken_heart: [Build #282520](https://buildkite.com/elastic/kibana-pull-request/builds/282520) failed c285e316af12c2c1c0e3cac8773ee86a1e241316
* :broken_heart: [Build #282495](https://buildkite.com/elastic/kibana-pull-request/builds/282495) failed d46d96a82e3a439d04b3159d207c6c6982b296d7
* :broken_heart: [Build #282431](https://buildkite.com/elastic/kibana-pull-request/builds/282431) failed d46d96a82e3a439d04b3159d207c6c6982b296d7
* :broken_heart: [Build #282319](https://buildkite.com/elastic/kibana-pull-request/builds/282319) failed 1712bce1b208bfff5f9b550deee5998f755775a3
* :broken_heart: [Build #282295](https://buildkite.com/elastic/kibana-pull-request/builds/282295) failed 7fbc38dc54b669e08e48123f1a21342578dd9893

cc @nchaulet

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/282770","number":282770,"commit":"ac6bd5af605d1fe03706cf4ef84c06b976b835a1"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/282520","number":282520,"commit":"c285e316af12c2c1c0e3cac8773ee86a1e241316"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/282495","number":282495,"commit":"d46d96a82e3a439d04b3159d207c6c6982b296d7"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/282431","number":282431,"commit":"d46d96a82e3a439d04b3159d207c6c6982b296d7"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/282319","number":282319,"commit":"1712bce1b208bfff5f9b550deee5998f755775a3"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/282295","number":282295,"commit":"7fbc38dc54b669e08e48123f1a21342578dd9893"}],"number":282770}
buildkite-pr-comment-->

nchaulet committed:
refacto upgrade to no support multiple policies

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/282918)
* Commit: 3715d80efb610b5c2b75acd2d086a1ef736f487e

### Failed CI Steps
* [FTR Configs #97](https://buildkite.com/elastic/kibana-pull-request/builds/282918#01958656-fb29-4801-aaeb-2289e5af8fba)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/282918#01958656-fb29-4801-aaeb-2289e5af8fba) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/282918/jobs/01958656-fb29-4801-aaeb-2289e5af8fba/artifacts/01958679-9468-48c2-9bdb-6df7d29d58be) FTR Configs #<span></span>97 / discover/group1 discover accessibility top nav menu buttons should return focus to the alerts button when dismissing the alerts popover

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Public APIs missing comments
> Total count of every public API that lacks a comment. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats comments` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/e0bf98e45a6c2b4e138cac601c6082ca68fb8301) | [after](https://github.com/elastic/kibana/commit/3715d80efb610b5c2b75acd2d086a1ef736f487e) | diff |
| --- | --- | --- | --- |
| `fleet` | 1338 | 1346 | +8 |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/e0bf98e45a6c2b4e138cac601c6082ca68fb8301) | [after](https://github.com/elastic/kibana/commit/3715d80efb610b5c2b75acd2d086a1ef736f487e) | diff |
| --- | --- | --- | --- |
| `fleet` | 1466 | 1474 | +8 |


</details>


### History
* :green_heart: [Build #282770](https://buildkite.com/elastic/kibana-pull-request/builds/282770) succeeded ac6bd5af605d1fe03706cf4ef84c06b976b835a1
* :broken_heart: [Build #282520](https://buildkite.com/elastic/kibana-pull-request/builds/282520) failed c285e316af12c2c1c0e3cac8773ee86a1e241316
* :broken_heart: [Build #282495](https://buildkite.com/elastic/kibana-pull-request/builds/282495) failed d46d96a82e3a439d04b3159d207c6c6982b296d7
* :broken_heart: [Build #282431](https://buildkite.com/elastic/kibana-pull-request/builds/282431) failed d46d96a82e3a439d04b3159d207c6c6982b296d7
* :broken_heart: [Build #282319](https://buildkite.com/elastic/kibana-pull-request/builds/282319) failed 1712bce1b208bfff5f9b550deee5998f755775a3
* :broken_heart: [Build #282295](https://buildkite.com/elastic/kibana-pull-request/builds/282295) failed 7fbc38dc54b669e08e48123f1a21342578dd9893

cc @nchaulet

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/282918","number":282918,"commit":"3715d80efb610b5c2b75acd2d086a1ef736f487e"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/282770","number":282770,"commit":"ac6bd5af605d1fe03706cf4ef84c06b976b835a1"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/282520","number":282520,"commit":"c285e316af12c2c1c0e3cac8773ee86a1e241316"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/282495","number":282495,"commit":"d46d96a82e3a439d04b3159d207c6c6982b296d7"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/282431","number":282431,"commit":"d46d96a82e3a439d04b3159d207c6c6982b296d7"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/282319","number":282319,"commit":"1712bce1b208bfff5f9b550deee5998f755775a3"}],"number":282918}
buildkite-pr-comment-->

nchaulet commented:
@elasticmachine merge upstream

elasticmachine committed:
Merge branch 'main' into feature-bulk-upgrade

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/282980)
* Commit: 29919acac48b83e1969b65e82056a9bf543d3c4d

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Public APIs missing comments
> Total count of every public API that lacks a comment. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats comments` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/323cbdb9cdfed351445a9e24523b78b8fd23dbb5) | [after](https://github.com/elastic/kibana/commit/29919acac48b83e1969b65e82056a9bf543d3c4d) | diff |
| --- | --- | --- | --- |
| `fleet` | 1338 | 1346 | +8 |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/323cbdb9cdfed351445a9e24523b78b8fd23dbb5) | [after](https://github.com/elastic/kibana/commit/29919acac48b83e1969b65e82056a9bf543d3c4d) | diff |
| --- | --- | --- | --- |
| `fleet` | 1466 | 1474 | +8 |


</details>


### History
* :yellow_heart: [Build #282918](https://buildkite.com/elastic/kibana-pull-request/builds/282918) was flaky 3715d80efb610b5c2b75acd2d086a1ef736f487e
* :green_heart: [Build #282770](https://buildkite.com/elastic/kibana-pull-request/builds/282770) succeeded ac6bd5af605d1fe03706cf4ef84c06b976b835a1
* :broken_heart: [Build #282520](https://buildkite.com/elastic/kibana-pull-request/builds/282520) failed c285e316af12c2c1c0e3cac8773ee86a1e241316
* :broken_heart: [Build #282495](https://buildkite.com/elastic/kibana-pull-request/builds/282495) failed d46d96a82e3a439d04b3159d207c6c6982b296d7
* :broken_heart: [Build #282431](https://buildkite.com/elastic/kibana-pull-request/builds/282431) failed d46d96a82e3a439d04b3159d207c6c6982b296d7
* :broken_heart: [Build #282319](https://buildkite.com/elastic/kibana-pull-request/builds/282319) failed 1712bce1b208bfff5f9b550deee5998f755775a3

cc @nchaulet

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/282980","number":282980,"commit":"29919acac48b83e1969b65e82056a9bf543d3c4d"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/282918","number":282918,"commit":"3715d80efb610b5c2b75acd2d086a1ef736f487e"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/282770","number":282770,"commit":"ac6bd5af605d1fe03706cf4ef84c06b976b835a1"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/282520","number":282520,"commit":"c285e316af12c2c1c0e3cac8773ee86a1e241316"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/282495","number":282495,"commit":"d46d96a82e3a439d04b3159d207c6c6982b296d7"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/282431","number":282431,"commit":"d46d96a82e3a439d04b3159d207c6c6982b296d7"}],"number":282980}
buildkite-pr-comment-->

jen-huang commented on x-pack/platform/plugins/shared/fleet/server/services/epm/agent/agent.ts:
```diff
@@ -79,7 +90,6 @@ function buildTemplateVariables(logger: Logger, variables: PackagePolicyConfigRe
     // support variables with . like key.patterns
     const keyParts = key.split('.');
     const lastKeyPart = keyParts.pop();
-    logger.debug(`Building agent template variables`);
```
nit: should this logger be removed?

nchaulet replied:
when debugging I found that log pretty verbose and I am not sure it bring so much value

jen-huang commented on x-pack/platform/plugins/shared/fleet/server/services/epm/packages/cache.ts:
```diff
@@ -52,6 +57,25 @@ class CacheSession {
     return this._agentTemplateAssetsMap;
   }
+
+  getHandlebarsCompiledTemplateCache() {
```
worth adding some test for these changes in `cache.test.ts`?

nchaulet replied:
Added in https://github.com/elastic/kibana/pull/213587/commits/a75245719e1d2d3543d353a7719a030b4aec7563

jen-huang commented on x-pack/platform/plugins/shared/fleet/server/services/package_policy.ts:
```diff
@@ -1702,47 +1775,173 @@ class PackagePolicyClientImpl implements PackagePolicyClient {
+                upgradePkgVersion
+              );
+
+            if (currentPackagePolicy.is_managed && !options?.force) {
+              throw new PackagePolicyRestrictionRelatedError(`Cannot upgrade package policy ${id}`);
+            }
```
I see this same check is done in the nonbulk `upgrade()` function too. do you think we could move the check to `doUpgrade()`?

nchaulet replied:
fixed in https://github.com/elastic/kibana/pull/213587/commits/579ee9bc293ce1e3f6834cd7b3c9abfd7c93896f

juliaElastic commented on x-pack/platform/plugins/shared/fleet/server/services/epm/packages/cache.ts:
```diff
@@ -26,6 +27,10 @@ class CacheSession {
 
   private _agentTemplateAssetsMap?: LRUCache<string, PackagePolicyAssetsMap>;
 
+  private _handlebarsCompiledTemplate?: LRUCache<string, HandlebarsTemplateDelegate>;
```
we should check how much memory does this cache use when full

nchaulet replied:
It seems pretty negligible, when running aws upgrade, that is our package with the most of agent template the memory usage was almost the same as before.

gergoabraham reviewed (approved):
@elastic/security-defend-workflows related change looks good 👍 
```
x-pack/platform/plugins/shared/osquery/server/routes/status/create_status_route.ts
```

nchaulet commented on x-pack/platform/plugins/shared/fleet/server/services/package_policy.ts:
```diff
@@ -1702,47 +1775,173 @@ class PackagePolicyClientImpl implements PackagePolicyClient {
     return runWithCache(async () => {
       const result: UpgradePackagePolicyResponse = [];
+      for (const chunkedIds of chunk(MAX_CONCURRENT_AGENT_POLICIES_OPERATIONS, ids)) {
+        const packagePolicies = await this.getByIDs(soClient, chunkedIds, {
```
Looking into and I think it probably make sense to move the whole upgrade/bulkUpgrade to their own module in `services/package_policy/upgrade` I will refactor this

nchaulet committed:
Refacto after code review

nchaulet commented:
@juliaElastic @jen-huang just pushed a big commit https://github.com/elastic/kibana/pull/213587/commits/579ee9bc293ce1e3f6834cd7b3c9abfd7c93896f that is addressing comment and moving everything related to upgrade out of the huge package policy service to a dedicated module let me know what you think of that refacto

nchaulet committed:
add cache test

nchaulet committed:
deduplicate managed check

nchaulet commented on x-pack/platform/plugins/shared/fleet/server/services/package_policy_service.ts:
```diff
@@ -224,15 +235,6 @@ export interface PackagePolicyClient {
     request?: KibanaRequest
   ): Promise<void>;
 
-  getUpgradePackagePolicyInfo(
```
was only here for testing as an internal of package policy service

nchaulet committed:
fix import

nchaulet committed:
fix typo in do upgrade

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/283880)
* Commit: a12bdec6ddd74c2b9359c5fb233446504bb11fdb

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Public APIs missing comments
> Total count of every public API that lacks a comment. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats comments` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/323cbdb9cdfed351445a9e24523b78b8fd23dbb5) | [after](https://github.com/elastic/kibana/commit/a12bdec6ddd74c2b9359c5fb233446504bb11fdb) | diff |
| --- | --- | --- | --- |
| `fleet` | 1338 | 1345 | +7 |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/323cbdb9cdfed351445a9e24523b78b8fd23dbb5) | [after](https://github.com/elastic/kibana/commit/a12bdec6ddd74c2b9359c5fb233446504bb11fdb) | diff |
| --- | --- | --- | --- |
| `fleet` | 1466 | 1473 | +7 |


</details>


### History
* :green_heart: [Build #282980](https://buildkite.com/elastic/kibana-pull-request/builds/282980) succeeded 29919acac48b83e1969b65e82056a9bf543d3c4d
* :yellow_heart: [Build #282918](https://buildkite.com/elastic/kibana-pull-request/builds/282918) was flaky 3715d80efb610b5c2b75acd2d086a1ef736f487e
* :green_heart: [Build #282770](https://buildkite.com/elastic/kibana-pull-request/builds/282770) succeeded ac6bd5af605d1fe03706cf4ef84c06b976b835a1
* :broken_heart: [Build #282520](https://buildkite.com/elastic/kibana-pull-request/builds/282520) failed c285e316af12c2c1c0e3cac8773ee86a1e241316
* :broken_heart: [Build #282495](https://buildkite.com/elastic/kibana-pull-request/builds/282495) failed d46d96a82e3a439d04b3159d207c6c6982b296d7
* :broken_heart: [Build #282431](https://buildkite.com/elastic/kibana-pull-request/builds/282431) failed d46d96a82e3a439d04b3159d207c6c6982b296d7

cc @nchaulet

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/283880","number":283880,"commit":"a12bdec6ddd74c2b9359c5fb233446504bb11fdb"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/282980","number":282980,"commit":"29919acac48b83e1969b65e82056a9bf543d3c4d"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/282918","number":282918,"commit":"3715d80efb610b5c2b75acd2d086a1ef736f487e"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/282770","number":282770,"commit":"ac6bd5af605d1fe03706cf4ef84c06b976b835a1"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/282520","number":282520,"commit":"c285e316af12c2c1c0e3cac8773ee86a1e241316"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/282495","number":282495,"commit":"d46d96a82e3a439d04b3159d207c6c6982b296d7"}],"number":283880}
buildkite-pr-comment-->

jen-huang reviewed (approved):
Thanks for the changes! Code LGTM 🚀

juliaElastic reviewed (approved):
LGTM

nchaulet commented:
@elasticmachine merge upstream

elasticmachine committed:
Merge branch 'main' into feature-bulk-upgrade

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/284090)
* Commit: 37e24fbbb079556d0fed9adfa3ae0558c3f765ef

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Public APIs missing comments
> Total count of every public API that lacks a comment. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats comments` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/21ab897e17d1397ed094f9a75790011bdbe666b6) | [after](https://github.com/elastic/kibana/commit/37e24fbbb079556d0fed9adfa3ae0558c3f765ef) | diff |
| --- | --- | --- | --- |
| `fleet` | 1338 | 1345 | +7 |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/21ab897e17d1397ed094f9a75790011bdbe666b6) | [after](https://github.com/elastic/kibana/commit/37e24fbbb079556d0fed9adfa3ae0558c3f765ef) | diff |
| --- | --- | --- | --- |
| `fleet` | 1466 | 1473 | +7 |


</details>


### History
* :green_heart: [Build #283880](https://buildkite.com/elastic/kibana-pull-request/builds/283880) succeeded a12bdec6ddd74c2b9359c5fb233446504bb11fdb
* :green_heart: [Build #282980](https://buildkite.com/elastic/kibana-pull-request/builds/282980) succeeded 29919acac48b83e1969b65e82056a9bf543d3c4d
* :yellow_heart: [Build #282918](https://buildkite.com/elastic/kibana-pull-request/builds/282918) was flaky 3715d80efb610b5c2b75acd2d086a1ef736f487e
* :green_heart: [Build #282770](https://buildkite.com/elastic/kibana-pull-request/builds/282770) succeeded ac6bd5af605d1fe03706cf4ef84c06b976b835a1
* :broken_heart: [Build #282520](https://buildkite.com/elastic/kibana-pull-request/builds/282520) failed c285e316af12c2c1c0e3cac8773ee86a1e241316
* :broken_heart: [Build #282495](https://buildkite.com/elastic/kibana-pull-request/builds/282495) failed d46d96a82e3a439d04b3159d207c6c6982b296d7

cc @nchaulet

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/284090","number":284090,"commit":"37e24fbbb079556d0fed9adfa3ae0558c3f765ef"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/283880","number":283880,"commit":"a12bdec6ddd74c2b9359c5fb233446504bb11fdb"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/282980","number":282980,"commit":"29919acac48b83e1969b65e82056a9bf543d3c4d"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/282918","number":282918,"commit":"3715d80efb610b5c2b75acd2d086a1ef736f487e"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/282770","number":282770,"commit":"ac6bd5af605d1fe03706cf4ef84c06b976b835a1"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/282520","number":282520,"commit":"c285e316af12c2c1c0e3cac8773ee86a1e241316"}],"number":284090}
buildkite-pr-comment-->

nchaulet commented:
@elasticmachine merge upstream

elasticmachine committed:
Merge branch 'main' into feature-bulk-upgrade

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/284400)
* Commit: 85b8976b7a271ac2310293dc3c9e947de4d47456

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Public APIs missing comments
> Total count of every public API that lacks a comment. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats comments` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/a9352bb57287cc6a06da99067b5dea694786e212) | [after](https://github.com/elastic/kibana/commit/85b8976b7a271ac2310293dc3c9e947de4d47456) | diff |
| --- | --- | --- | --- |
| `fleet` | 1338 | 1345 | +7 |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/a9352bb57287cc6a06da99067b5dea694786e212) | [after](https://github.com/elastic/kibana/commit/85b8976b7a271ac2310293dc3c9e947de4d47456) | diff |
| --- | --- | --- | --- |
| `fleet` | 1466 | 1473 | +7 |


</details>


### History
* :green_heart: [Build #284090](https://buildkite.com/elastic/kibana-pull-request/builds/284090) succeeded 37e24fbbb079556d0fed9adfa3ae0558c3f765ef
* :green_heart: [Build #283880](https://buildkite.com/elastic/kibana-pull-request/builds/283880) succeeded a12bdec6ddd74c2b9359c5fb233446504bb11fdb
* :green_heart: [Build #282980](https://buildkite.com/elastic/kibana-pull-request/builds/282980) succeeded 29919acac48b83e1969b65e82056a9bf543d3c4d
* :yellow_heart: [Build #282918](https://buildkite.com/elastic/kibana-pull-request/builds/282918) was flaky 3715d80efb610b5c2b75acd2d086a1ef736f487e
* :green_heart: [Build #282770](https://buildkite.com/elastic/kibana-pull-request/builds/282770) succeeded ac6bd5af605d1fe03706cf4ef84c06b976b835a1
* :broken_heart: [Build #282520](https://buildkite.com/elastic/kibana-pull-request/builds/282520) failed c285e316af12c2c1c0e3cac8773ee86a1e241316

cc @nchaulet

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/284400","number":284400,"commit":"85b8976b7a271ac2310293dc3c9e947de4d47456"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/284090","number":284090,"commit":"37e24fbbb079556d0fed9adfa3ae0558c3f765ef"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/283880","number":283880,"commit":"a12bdec6ddd74c2b9359c5fb233446504bb11fdb"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/282980","number":282980,"commit":"29919acac48b83e1969b65e82056a9bf543d3c4d"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/282918","number":282918,"commit":"3715d80efb610b5c2b75acd2d086a1ef736f487e"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/282770","number":282770,"commit":"ac6bd5af605d1fe03706cf4ef84c06b976b835a1"}],"number":284400}
buildkite-pr-comment-->

whyyouwannaknow commented:
Hello,

Do we know if this will be backported to any 8.X version? Or do we will have to wait for the release of the 9.1.0 to get this improvement?

Thank you,

kpollich commented:
Yeah unfortunately this is a pretty high impact set of changes that's a bit difficult to justify backporting to an 8.x minor.
